marklogic-data-hub/src/test/java/com/marklogic/hub/HubTestBase.java

```diff
@@ -946,6 +969,13 @@ protected void installUserModules(HubConfig hubConfig, boolean force) {
 
         LoadModulesCommand loadModulesCommand = new LoadModulesCommand();
         commands.add(loadModulesCommand);
+        //Call 'generateFunctionMetadataCommand' if mapping functions are deployed. If mapping is deployed first
+        // and it references custom functions, mapping xslt will not have reference to custom functions and tests
+        // would fail with XDMP-UNDFUN: (err:XPST0017) Undefined function customDateTime().
+
+        if(hubConfig.getHubProject().getCustomModulesDir().resolve("mapping-functions").toFile().exists()){
+            commands.add(generateFunctionMetadataCommand);
```

Comment:
Is there any downside to always adding this? I think it's fine to always add it. Otherwise, future tests may not work because they don't meet the condition of the "if" statement.

Reply:
Ok, will include that.. no downside except a few a delay of few ms every time this method is called